flink-table/flink-table-common/src/main/java/org/apache/flink/table/factories/TableSinkFactory.java

```diff
@@ -53,4 +55,39 @@
 		return createTableSink(table.toProperties());
 	}
 
+	/**
+	 * Creates and configures a {@link TableSink} based on the given
+	 {@link Context}.
+	 *
+	 * @param context context of this table sink.
+	 * @return the configured table sink.
+	 */
+	default TableSink<T> createTableSink(Context context) {
+		return createTableSink(
+				context.getObjectIdentifier().toObjectPath(),
+				context.getTable());
+	}
+
+	/**
+	 * Context of table sink creation. Contains table information and
+	 environment information.
+	 */
+	interface Context {
+
+		/**
+		 * @return full identifier of the given {@link CatalogTable}.
+		 */
+		ObjectIdentifier getObjectIdentifier();
+
+		/**
+		 * @return table {@link CatalogTable} instance.
+		 */
+		CatalogTable getTable();
+
+		/**
+		 * @return readable config of this table environment.
```

Comment:
Add a description that the configuration gives the factory instance the ability to access `TableConfig#getConfiguration()` which holds the current TableEnvironment session configurations.